File: Python/7py 2/sequence_sum.py
"""
|--------------------------------------------------------------------------
| Sum of a sequence
|--------------------------------------------------------------------------
|
"""

def sequence_sum(begin_number, end_number, step):
   res = 0
   for i in range(begin_number, end_number+1, step):
      res += i
   return res


def sequence_sum(start, end, step):
   return sum(range(start, end+1, step))


def sequence_sum(b, e, s):
   k = (e - b) // s
   return (1 + k) * (b + s * k / 2) if b <= e else 0


def sequence_sum(beg, end, step):
   if beg > end:
      return 0
   return beg + sequence_sum(beg+step, end, step)


print(sequence_sum(2, 6, 2), 12)
print(sequence_sum(1, 5, 1), 15)
print(sequence_sum(1, 5, 3), 5)
print(sequence_sum(16, 15, 3), 0)

File: JavaScript/7js 3/sequenceSum.js
/*
|--------------------------------------------------------------------------
| Sum of a sequence
|--------------------------------------------------------------------------
|
*/

function sequenceSum(begin, end, step) {
   res = 0;
   for (let i = begin; i <= end; i += step) {
      res += i;
   }
   return res;
}


const sequenceSum = (begin, end, step) => {
   let n = Math.floor((end - begin) / step + 1);
   return n <= 0 ? 0 : n * (begin + (begin + (n - 1) * step)) / 2;
};


const sequenceSum = (begin, end, step) => {
   return begin > end ? 0 : begin + sequenceSum(begin + step, end, step)
};


const sequenceSum = (a, b, c) => (
   k => (k + 1) * (2 * a + c * k) / 2
)(a > b ? -1 : (b - a) / c | 0);


console.log(sequenceSum(2, 6, 2), 12)
console.log(sequenceSum(1, 5, 1), 15)
console.log(sequenceSum(1, 5, 3), 5)